qmi/core/transport.py

```diff
@@ -647,18 +630,38 @@ def read(self, nbytes: int, timeout: Optional[float]) -> bytes:
         while nbuf < nbytes:
             self._safe_socket.settimeout(tremain)
             try:
-                b = self._safe_socket.recv(nbytes - nbuf)
+                if self._assert_addr:
+                    # For UDP, reading data fails if the arriving packet size is larger than the given nbytes (- nbuf)
+                    # to read. To avoid this happening, read at least 4096 standard that is larger than regular MTU.
+                    udp_packet_size = max(nbytes - nbuf, 4096)
+                    try:
+                        b, addr = self._safe_socket.recvfrom(udp_packet_size)
+                    except (BlockingIOError, socket.timeout) as err:
+                        raise err  # Re-raise this for handling in the upper try-except.
+                    except OSError:
```

Comment:
just like the previous exception, it would be good to re raise this one as `except OSError as exc` and then `raise QMI_RuntimeException(...) from exc`

Reply:
Will do